File: java/1047.java
class Solution {
    public String removeDuplicates(String S) {
        Stack<Character> stack = new Stack();
        for(int i = 0;i<S.length();i++){
            if(stack.isEmpty() || stack.peek() != S.charAt(i)) {
				stack.add(S.charAt(i));
			}else {
				stack.pop();
			}
        }
        StringBuilder ans = new StringBuilder();
        for(char c:stack){
            ans.append(c);
        }
    return ans.toString();
    }
}

File: go/1047.go
func removeDuplicates(S string) string {
	bs := []byte(S)
	stack, top := make([]byte, len(bs)), -1
	for i := 0; i < len(S); i++ {
		if top >= 0 && stack[top] == S[i] {
			top--
		} else {
			top++
			stack[top] = S[i]
		}
	}
	return string(stack[:top+1])
}